### ISSUE TITLE
[Extension Proposal] SQLDelight

### ISSUE THREAD

**Issue by x80486** [quarkusio/quarkus#35074]
### Description

`SQLDelight` generates typesafe Kotlin APIs from SQL statements. This is an interesting alternative to some other frameworks/libraries because it's like a supercharged JDBC version.

Not sure how the code generated might work with Java, but since Kotlin is 100% interoperable with Java, it should work.SQLDelight generates typesafe Kotlin APIs from your SQL statements

### Repository name

quarkus-sqldelight

### Short description

SQLDelight generates typesafe Kotlin APIs from SQL statements

### Repository Homepage URL

https://quarkiverse.github.io/quarkiverse-docs/<REPOSITORY_NAME>/dev/

### Repository Topics

- database-access
- quarkus-extension
- sqldelight
- sql

**Comment by gastaldi** [comment]
Browsing the website it seems that SQLDelight is published as Gradle Plugin and also supported by IDE plugins. 

Can you elaborate on what would this extension do that would be more appealing than using the beforementioned plugins directly?

**Comment by x80486** [comment]
I was under the assumption that somehow frameworks/libraries need to be _integrated_ with `Quarkus` in order to behave correctly when compiled into a binary.

If that's not the case, I agree as well that `SQLDelight` could be used with the provided plugins, etc.

**Comment by gastaldi** [comment]
> I was under the assumption that somehow frameworks/libraries need to be _integrated_ with `Quarkus` in order to behave correctly when compiled into a binary.

Not necessarily. It depends on what you mean by _integrated_. For example, if you mean regenerating sources during `quarkus:dev` after performing a change in a resource or performing the necessary changes to make it compatible with native generation (if the library performs something that isn't allowed during native compilation), then yes, an extension makes sense.

**Comment by x80486** [comment]
Exactly. I don't know the details and what might be required, if anything...hence the ticket to see if it makes sense, or not.

**Comment by gastaldi** [comment]
I don't think there is enough buy-in for this extension, so I'm closing for now. Let's reopen if someone is interested in leading this
